mirascope/beta/realtime/openai/realtime.py

```diff
@@ -0,0 +1,255 @@
+                # print(f"Received message: {message}")
+                match message["type"]:  # type:
+                    case "session.created":
+                        if not self.vda_mode:
+                            await self._create_session_update(conn)
+                    case "response.audio.done":
+                        audio = audio_chunk_to_audio_segment(audio_chunk)
+                        await self._process_specific_receiver(audio, "audio")
+                        audio_chunk = b""
+                    case "response.audio.delta":
+                        audio_chunk += base64.b64decode(message["delta"])
+                        await self._process_specific_receiver(message, "audio_chunk")
+                    case "response.text.done":
+                        await self._process_specific_receiver(text_chunk, "text")
+                        text_chunk = ""
+                    case "response.text.delta":
+                        text_chunk += message["delta"]
+                        await self._process_specific_receiver(message, "text_chunk")
+                    case "response.audio_transcript.done":
+                        await self._process_specific_receiver(
+                            audio_transcript_chunk, "audio_transcript"
+                        )
+                        audio_transcript_chunk = ""
+                    case "response.audio_transcript.delta":
+                        audio_transcript_chunk += message["delta"]
+                        await self._process_specific_receiver(
+                            audio_transcript_chunk, "audio_transcript_chunk"
+                        )
+
+                    # TODO: Add more cases like "response.function_call_arguments.done"
+
+                # Call the receivers for all messages
+                for receiver in self.receivers[None]:
+                    await receiver(message, self.context)
+        except asyncio.CancelledError:
+            print("Receiver has been cancelled")
+        finally:
+            print("Receiver has been closed")
+
+    async def run(self) -> None:
+        if self.is_running():
+            raise RuntimeError("Already running")
+
+        headers = {
+            "Authorization": f"Bearer {os.environ["OPENAI_API_KEY"]}",
+            "OpenAI-Beta": "realtime=v1",
+        }
+
+        async for conn in connect(
+            uri=f"wss://api.openai.com/v1/realtime?model={self.model}",
+            additional_headers=headers,
+        ):
+            try:
+                receiver_task = asyncio.create_task(self._process_receiver(conn))
+                sender_task = asyncio.create_task(self._process_sender(conn))
+                await asyncio.gather(receiver_task, sender_task)
+            except websockets.ConnectionClosed:
+                continue
+
+    def is_running(self) -> bool:
+        return self._connection is not None
+
+
+if __name__ == "__main__":
```

Comment:
this should live under `exaples/beta/openai/realtime` or something like this rather than directly in the file

this will also be useful for writing the `Beta` section of our docs so people can see how to use this